src/coordination/raft_state.cpp

```diff
@@ -282,7 +316,7 @@ auto RaftState::AddCoordinatorInstance(CoordinatorToCoordinatorConfig const &con
   while (!maybe_stop()) {
     std::this_thread::sleep_for(std::chrono::milliseconds(waiting_period));
     const auto server_config = raft_server_->get_srv_config(static_cast<nuraft::int32>(config.coordinator_id));
-    if (server_config) {
+    if (server_config->get_id() == config.coordinator_id) {
```

Comment:
do we need to check this? aren't we searching via the id

Reply:
No you are right